### DISCUSSION TITLE
How do I specify the --turbo for @nrwl/next project?

### DISCUSSION THREAD

**Discussion by laneme** [nrwl/nx#13146] [General]
I dont have good understnading of nx and the configs/generators it provides. And i'm trying to use the --turbo config option next js recently introduced with next 13. I already upgraded all my next, nx & @nrwl/* packages but cant seem to make --turbo option work.

I tried doing it in the project.json like this `"buildTarget": "apps-prjam:build:development --turbo"`, doesnt seem to do anything. Can anyone help with this?

**Comment by jonsoku-dev** [comment]
same question

**Comment by ianldgs** [comment]
After digging through the code of NX and next a bit, what I came up with was:

`nx g custom-server mynextapp`

mynextapp/next.config.js:
```diff
+  experimental: {
+    appDir: true,
+  },
```

mynextapp/project.json:
```diff
        "production": {
          "buildTarget": "mynextapp:build:production",
-          "dev": false,
-          "customServerTarget": "mynextapp:serve-custom-server:production"
        }
```

mynextapp/server/main.ts:
```ts
import { workspaceRoot } from "@nrwl/devkit";
import { loadBindings } from "next/dist/build/swc";
import * as path from "path";

const dir = process.env.NX_NEXT_DIR || path.join(__dirname, "..");
const dev = process.env.NODE_ENV === "development";

const hostname = process.env.HOST;
const port = process.env.PORT ? parseInt(process.env.PORT) : 4200;

const devServerOptions = {
  allowRetry: true,
  dev,
  dir,
  hostname,
  isNextDevCommand: true,
  port,
};

async function main() {
  const bindings: any = await loadBindings();

  const server = bindings.turbo.startDev({
    ...devServerOptions,
    showAll: true,
    root: workspaceRoot,
  });

  return server;
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
```

**Comment by jaysoo** [comment]
You can do this now with the `--turbo` CLI option, or specify `turbo: true` for your serve target.

https://nx.dev/nx-api/next/executors/server#turbo

**Reply by ibufu** [reply]
Hi@jaysoo ,
May I know how to specify turbo for dev?
```json
 {
  "plugin": "@nx/next/plugin",
  "options": {
    "startTargetName": "start",
    "buildTargetName": "build",
    "devTargetName": "dev",
    "serveStaticTargetName": "serve-static"
   }
 }
```
`nx dev x-ui --turbo` works for me. But I want to turn no `turbo` by default.

**Reply by aholthagerty** [reply]
Im curious about this, too. running `nx dev my-app --turbo` is fine, but I want to specify in the next plugin configuration that it should use `--turbo` by default, imo.
